patches/amd-mainline/HIPIFY/0003-Don-t-override-CMAKE_C-XX-_COMPILER-on-MSVC.patch

```diff
@@ -0,0 +1,38 @@
+From bc55217782f5022b82f45a18fb1cae063c16d493 Mon Sep 17 00:00:00 2001
+From: Scott <scott.todd0@gmail.com>
+Date: Tue, 18 Feb 2025 15:54:03 -0800
+Subject: [PATCH 3/3] Don't override CMAKE_C[XX]_COMPILER on MSVC.
+
+```
+[build] [hipify] [38/39   2% :: 0.119] Building CXX object CMakeFiles\hipify-clang.dir\src\CUDA2HIP.cpp.obj
+[build] [hipify] FAILED: CMakeFiles/hipify-clang.dir/src/CUDA2HIP.cpp.obj
+[build] [hipify] ccache D:\projects\TheRock\build\compiler\amd-llvm\dist\lib\llvm\bin\clang++  /nologo /TP -DGTEST_HAS_RTTI=0 -D_HAS_EXCEPTIONS=0 -ID:\projects\TheRock\build\compiler\amd-llvm\dist\lib\llvm\include /DWIN32;/D_WINDOWS;/EHsc /DWIN32 /D_WINDOWS /GR /EHsc  -DLIB_CLANG_RES=19  /Zi /O2 /Ob1 /DNDEBUG -MD   -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -DUNICODE -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  /EHs-c- /GR- /std:c++17 /Od /GR- /EHs- /EHc- /MP /Zc:preprocessor /showIncludes /FoCMakeFiles\hipify-clang.dir\src\CUDA2HIP.cpp.obj /FdCMakeFiles\hipify-clang.dir\ /FS -c D:\projects\TheRock\compiler\hipify\src\CUDA2HIP.cpp
+[build] [hipify] clang++: error: no such file or directory: '/nologo'
+[build] [hipify] clang++: error: no such file or directory: '/TP'
+[build] [hipify] clang++: error: no such file or directory: '/DWIN32;/D_WINDOWS;/EHsc'
+[build] [hipify] clang++: error: no such file or directory: '/DWIN32'
+[build] [hipify] clang++: error: no such file or directory: '/D_WINDOWS'
+[build] [hipify] clang++: error: no such file or directory: '/GR'
+```
+---
+ CMakeLists.txt | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index dad79610..f3616fbc 100755
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -115,6 +115,10 @@ if (NOT HIPIFY_CLANG_TESTS_ONLY)
+           ${LLVM_BINARY_DIR}/tools/lld/include
+           ${LLVM_EXTERNAL_LLD_SOURCE_DIR}/include)
+     endif()
++  elseif(MSVC)
++    # Keep the existing compiler since MSVC-specific flags will be set later.
++    # Overriding CMAKE_CXX_COMPILER/CMAKE_C_COMPILER below is sketchy - could
++    # that be removed or reworked instead?
+   else()
+     set(CMAKE_CXX_COMPILER ${LLVM_TOOLS_BINARY_DIR}/clang++)
+     set(CMAKE_C_COMPILER ${LLVM_TOOLS_BINARY_DIR}/clang)
```

Comment:
Some history for the code in question: https://github.com/ROCm/HIPIFY/pull/937. Some shaky assumptions there and in later patches about which configurations are built on each platform and with each compiler.

Reply:
I'm trying to reproduce my issues in the HIPIFY project and send a patch but the instructions (https://github.com/ROCm/HIPIFY/blob/amd-staging/docs/building/building-hipify.rst) are missing some important information for Windows. The build step seems to require a source build of LLVM, instructions only talk about the Visual Studio CMake generator, etc.

Reply:
Reproduced using the ninja generator, VS generator doesn't exhibit the issue. Filed https://github.com/ROCm/HIPIFY/issues/1864.